lib/iris/_merge.py

```diff
@@ -33,7 +33,8 @@
 import numpy as np
 import numpy.ma as ma
 
-from iris._lazy_data import as_lazy_data, is_lazy_data, multidim_lazy_stack
+from iris._lazy_data import (as_lazy_data, is_lazy_data, multidim_lazy_stack,
+                             as_concrete_data)
```

Comment:
@lbdreyer The sort order needs changes here ... make `as_concrete_data` first ...